packages/cf-deploy-config-inquirer/src/types.ts

```diff
@@ -0,0 +1,86 @@
+import type { YUIQuestion } from '@sap-ux/inquirer-common';
+import type { AutocompleteQuestionOptions } from 'inquirer-autocomplete-prompt';
+
+/**
+ * Enum defining prompt names for Cloud Foundry (CF) deployment configuration.
+ */
+export enum promptNames {
+    /** The prompt to specify the destination name for CF deployment. */
+    destinationName = 'destinationName',
+    /** The prompt to specify if a managed app router should be added to the deployment. */
+    addManagedAppRouter = 'addManagedAppRouter',
+    /** The prompt for confirming destination overwrite. */
+    overwrite = 'overwriteDestinationName'
+}
+
+/**
+ * Configuration options for the 'destinationName' prompt used in deployment settings.
+ */
+export type DestinationNamePromptOptions = {
+    /** The Cloud Foundry destination name to be used in the deployment process. */
+    destination?: string;
+    /** Default value to suggest for the destination name. */
+    defaultValue: string;
+    /** Flag to indicate if a hint message should be shown to indicate the app router is configured to use direct service binding . */
+    directBindingDestinationHint?: boolean;
+    /**
+     * List of additional destination choices available for the prompt.
+     * - In BAS environments, this list will be appended to BTP destination options.
+     * - If `additionalChoiceList` is provided and the environment is VS Code,
+     *   the prompt will render as a list, allowing users to select from the provided choices instead of input.
+     */
+    additionalChoiceList?: CfSystemChoice[];
+    /**
+     * Flag to indicate if the destination prompt should use auto completion
+     */
+    useAutocomplete?: boolean;
+};
+
+/**
+ * Defines options for boolean-type prompts in CF deployment configuration.
+ */
+type booleanValuePromptOptions = Record<promptNames.overwrite, boolean> &
+    Record<promptNames.addManagedAppRouter, boolean>;
+
+/**
+ * Defines options for string-type prompts in CF deployment configuration.
+ */
+type stringValuePromptOptions = Record<promptNames.destinationName, DestinationNamePromptOptions>;
+
+/**
+ * Configuration options for CF deployment prompts.
+ */
+export type CfDeployConfigPromptOptions = Partial<stringValuePromptOptions & booleanValuePromptOptions>;
+
+/**
+ * Represents a question in the CF deployment configuration.
+ * Extends `YUIQuestion` with optional autocomplete functionality.
+ */
+export type CfDeployConfigQuestions = YUIQuestion<CfDeployConfigAnswers> &
+    Partial<Pick<AutocompleteQuestionOptions, 'source'>>;
+
+/**
+ * User responses for CF deployment configuration.
+ */
+export interface CfDeployConfigAnswers {
+    /** The selected Cloud Foundry destination. */
+    destination?: string;
```

Comment:
This doesn't look like a `destination` but a destination name?. Could we consider renaming to be clear as there is also a `Destination` type which encapsulates all destination properties.

Reply:
Renamed to `destinationName` 👍